### PR TITLE
Split Sponge x Merkle

### LINKED ISSUE #31: Apply `rustfmt` to all sources

### PR DESCRIPTION
**Pull request by vlopes11**
Changes:
* Applied `rustfmt` (fixes #31 )
* Separated the merkle hash from the sponge hash

TODO:
* Implement both merkle and sponge (will be done in different PRs)

### CODE REVIEW

File: src/scalar/mod.rs [hunk 1]
```diff
@@ -3,101 +3,103 @@ use crate::mds_matrix::MDS_MATRIX;
...
+    if data.len() >= width - 1 {
+        return Err(PermError::InputFull);
+    }
 
-  let mut words = vec![Scalar::zero(); width];
-  let words_slice = &mut words[1..1 + data.len()];
+    // The base type declares the output type, so we use u64
+    // since the arity of the merkle tree is not likely to be
+    // >= 2^64.
+    let first_item = Scalar::from((1u64 << width) - 1);
```
Comment (CPerezz) [inline src/scalar/mod.rs]: Same as above.

File: src/scalar/sponge.rs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    let width = MDS_MATRIX.len();
+
+    if data.len() >= width - 1 {
+        return Err(PermError::InputFull);
+    }
+
+    // The base type declares the output type, so we use u64
+    // since the arity of the merkle tree is not likely to be
+    // >= 2^64.
+    let first_item = Scalar::from((1u64 << width) - 1);
```
Comment (CPerezz) [inline src/scalar/sponge.rs]: Same as above.

File: src/linear_combination/sponge.rs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    let width = MDS_MATRIX.len();
+
+    if data.len() >= width - 1 {
+        return Err(PermError::InputFull);
+    }
+
+    // The base type declares the output type, so we use u64
+    // since the arity of the merkle tree is not likely to be
+    // >= 2^64.
+    let first_item = LinearCombination::from(Scalar::from((1u64 << width) - 1));
```
[the remark this answers is not in the document]
Comment (decentralisedkev) [reply src/linear_combination/sponge.rs]: @CPerezz I believe the first element is zero for the sponge construction, for the merkle tree variation; the first element is a `32 * 8` bit scalar. Where the n'th bit being one means that we have a non-empty leaf in that position.

**Comment by ZER0** [comment]
As discussed, this is not longer applicable.
Also, `rustfmt` would be done separately (see issue #31)

**Pull request closed without merging**